Rover/fence.cpp

```diff
@@ -1,26 +1,51 @@
 #include "Rover.h"
 
-// fence_check - ask fence library to check for breaches and initiate the response
-void Rover::fence_check()
-{
 #if AP_FENCE_ENABLED
-    uint8_t new_breaches;  // the type of fence that has been breached
+
+// async fence checking io callback at 1Khz
+void Rover::fence_run_checks()
+{
+    uint32_t now = AP_HAL::millis();
+
+    if (!AP_HAL::timeout_expired(fence_breaches.last_check_ms, now, 100U)) { // 10Hz update rate
+        return;
+    }
+
+    if (fence_breaches.have_updates) {
+        return; // wait for the main loop to pick up the new breaches before checking again
+    }
+
+    fence_breaches.last_check_ms = now;
     const uint8_t orig_breaches = fence.get_breaches();
+    // check for new breaches; new_breaches is bitmask of fence types breached
+    fence_breaches.new_breaches = fence.check();
 
-    // check for a breach
-    new_breaches = fence.check();
+    if (!fence_breaches.new_breaches && orig_breaches && fence.get_breaches() == 0) {
+        // record clearing of breach
+        LOGGER_WRITE_ERROR(LogErrorSubsystem::FAILSAFE_FENCE, LogErrorCode::ERROR_RESOLVED);
```

Comment:
```suggestion
        LOGGER_WRITE_ERROR(LogErrorSubsystem::FAILSAFE_FENCE, 0);
```

... the enumeration value should be considered opaque in terms of its value.

Reply:
This is not code I wrote, it pre-exists this PR - I'd really like to avoid scope-creeping PR's if possible.